services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/AwsConnectionServiceImpl.java

```diff
@@ -117,4 +121,19 @@ private AWSDatasource awsDatasourceMapper(AwsDatasourceRequest awsDatasourceRequ
         awsDatasource.setActive(true);
         return awsDatasource;
     }
+
+    @Override
+    public List<Bucket> testConnection(AwsDatasourceRequest awsDatasourceRequest) throws InvalidAccessTypeException {
+
+        AwsConnectionFactory connectionFactory = new AwsConnectionFactory();
```

Comment:
initiate object only when required.. why @autiwired is not used for this factory object